collect_app/src/main/java/org/odk/collect/android/activities/ViewFormHierarchyActivity.java

```diff
@@ -37,7 +38,11 @@
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        Collect.getInstance().getFormController().stepToOuterScreenEvent();
+        FormController formController = Collect.getInstance().getFormController();
+        if (formController == null) {
+            return;
```

Comment:
I don't think `return` is enough here, that means you just stop onCreate() method but the activity is still displayed, probably we should finish in such a case.

Reply:
I see how this is confusing. Finish is actually called in the super class based on the same condition. 

What I can do instead, is check here if the activity is actually finishing, by calling `isFinished()`. This will reduce code duplication and rely on the finishing condition.

Reply:
Ohh I get it so `return` seems fine.